finalitygadget/finalitygadget.go

```diff
@@ -91,13 +92,23 @@ func NewFinalityGadget(cfg *config.Config, db db.IDatabaseHandler, logger *zap.L
 		return nil, err
 	}
 
-	lastProcessedHeight := uint64(0)
-	latestBlock, err := db.QueryLatestFinalizedBlock()
-	if err != nil && !errors.Is(err, types.ErrBlockNotFound) {
-		return nil, fmt.Errorf("failed to query latest finalized block: %w", err)
+	// Query config from rollup-bsn contract to get bsn_activation_height
+	contractConfig, err := cwClient.QueryConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to query contract config: %w", err)
 	}
-	if latestBlock != nil {
-		lastProcessedHeight = latestBlock.BlockHeight
+
+	lastProcessedHeight := uint64(0)
+
+	// Check if configured start block height is valid against bsn_activation_height
+	if cfg.StartBlockHeight > 0 {
+		if cfg.StartBlockHeight < contractConfig.BsnActivationHeight {
+			return nil, fmt.Errorf("configured StartBlockHeight (%d) is less than bsn_activation_height (%d), earliest allowed block is %d",
+				cfg.StartBlockHeight, contractConfig.BsnActivationHeight, contractConfig.BsnActivationHeight)
+		}
+		lastProcessedHeight = cfg.StartBlockHeight - 1
+	} else {
+		return nil, fmt.Errorf("StartBlockHeight must be specified and >= bsn_activation_height (%d)", contractConfig.BsnActivationHeight)
 	}
```

Comment:
Hmm I see, so this `StartBlockHeight` is more like a local config specifying the height that FG wishes to start tallying. This makes sense to me, although we will need to document this a bit